### ISSUE TITLE
Regression in layout recognition (empty page)

### ISSUE THREAD

**Issue by stweil** [tesseract-ocr/tesseract#3404]
Since commit 5db92b2 and especially the modified [declaration for PartSetVector](https://github.com/tesseract-ocr/tesseract/commit/5db92b26aa4cab45f3da6714328c2fcd80891441#diff-652bff152d618e7512616aab32ae469997bfc933e1c1be86a1b8bf7dde4c5751L33) Tesseract no longer recognizes text on two images, but reports "Empty page".

See original JPEG files [470875348_0010](https://digi.bib.uni-mannheim.de/fileadmin/digi/470875348/max/470875348_0010.jpg) and [452117542_0250](https://digi.bib.uni-mannheim.de/fileadmin/digi/452117542/max/452117542_0250.jpg).

_Originally posted by @stweil in https://github.com/tesseract-ocr/tesseract/issues/707#issuecomment-823824074_

**Comment by stweil** [comment]
These functions are involved, so at least one of them shows wrong behaviour:
```
ColumnFinder::MakeColumns(bool single_column) # uses PartSetVector internally
ColumnFinder::ImproveColumnCandidates(PartSetVector *src_sets, PartSetVector *column_sets)
ColPartitionGrid::MakeColPartSets(PartSetVector *part_sets)
ColPartitionSet::ImproveColumnCandidate(WidthCallback cb, PartSetVector *src_sets)
ColPartitionSet::AddToColumnSetsIfUnique(PartSetVector *column_sets, WidthCallback cb)
```

**Comment by stweil** [comment]
The bug was in `ColumnFinder::ImproveColumnCandidates` and is fixed by commit 740d10b61be6009d4e523885b97168f8dd8c7a1c.
